robolectric/src/main/java/org/robolectric/internal/BuckManifestFactory.java

```diff
@@ -0,0 +1,77 @@
+package org.robolectric.internal;
+
+import org.robolectric.annotation.Config;
+import org.robolectric.manifest.AndroidManifest;
+import org.robolectric.res.Fs;
+import org.robolectric.res.FsFile;
+import org.robolectric.res.ResourcePath;
+import org.robolectric.util.Logger;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.ListIterator;
+
+public class BuckManifestFactory implements ManifestFactory {
+
+  private static final String BUCK_ROBOLECTRIC_RES_DIRECTORIES = "buck.robolectric_res_directories";
+  private static final String BUCK_ROBOLECTRIC_ASSETS_DIRECTORIES = "buck.robolectric_assets_directories";
```

Comment:
Wondering if it'd make sense to aim for build system agnosticism here? Maybe `robolectric.resources` and `robolectric.assets`? We're planning for a gradle plugin which could probably share properties... @jongerrish thoughts? Maybe this could be `PluginManifestFactory` or something?